test/e2e_node/runtime_conformance_test.go

```diff
@@ -17,122 +17,311 @@ limitations under the License.
 package e2e_node
 
 import (
+	"fmt"
+	"io/ioutil"
+	"os"
+	"path"
 	"time"
 
 	"k8s.io/kubernetes/pkg/api"
 	"k8s.io/kubernetes/pkg/client/restclient"
 	client "k8s.io/kubernetes/pkg/client/unversioned"
+	"k8s.io/kubernetes/pkg/util"
 
 	. "github.com/onsi/ginkgo"
 	. "github.com/onsi/gomega"
 )
 
 const (
-	retryTimeout = time.Minute * 4
+	retryTimeout = time.Minute * 5
 	pollInterval = time.Second * 5
 )
 
-var _ = Describe("Container Runtime Conformance Test", func() {
+type testStatus struct {
+	Name             string
+	RestartPolicy    api.RestartPolicy
+	Phase            api.PodPhase
+	State            ContainerState
+	RestartCountOper string
+	RestartCount     int32
+	Ready            bool
+}
+
+var _ = Describe("Container runtime Conformance Test", func() {
 	var cl *client.Client
 
 	BeforeEach(func() {
 		// Setup the apiserver client
 		cl = client.NewOrDie(&restclient.Config{Host: *apiServerAddress})
 	})
 
-	Describe("container conformance blackbox test", func() {
-		Context("when running a container that terminates", func() {
-			var containerCase ConformanceContainer
+	Describe("container runtime conformance blackbox test", func() {
+		namespace := "runtime-conformance"
+		Context("when start a container that exits success", func() {
+			var testCContainers []ConformanceContainer
 
 			BeforeEach(func() {
-				containerCase = ConformanceContainer{
-					Container: api.Container{
-						Image:           "gcr.io/google_containers/busybox",
-						Name:            "busybox",
-						Command:         []string{"sh", "-c", "env"},
-						ImagePullPolicy: api.PullIfNotPresent,
+				testCContainers = []ConformanceContainer{}
+			})
+
+			It("it should run with the expected status [Conformance]", func() {
+				testContainer := api.Container{
+					Image: "gcr.io/google_containers/busybox",
+					VolumeMounts: []api.VolumeMount{
+						{
+							MountPath: "/restart-count",
+							Name:      "restart-count",
+						},
 					},
-					Client:   cl,
-					Phase:    api.PodSucceeded,
-					NodeName: *nodeName,
+					ImagePullPolicy: api.PullIfNotPresent,
+				}
+				testVolumes := []api.Volume{
+					{
+						Name: "restart-count",
+						VolumeSource: api.VolumeSource{
+							HostPath: &api.HostPathVolumeSource{
+								Path: "/tmp",
+							},
+						},
+					},
+				}
+				testCount := int32(3)
+				testStatuss := []testStatus{
+					{"terminate-always", api.RestartPolicyAlways, api.PodRunning, ContainerStateWaiting | ContainerStateRunning | ContainerStateTerminated, ">", testCount, false},
+					{"terminate-onfailure", api.RestartPolicyOnFailure, api.PodSucceeded, ContainerStateTerminated, "==", testCount, false},
+					{"terminate-never", api.RestartPolicyNever, api.PodSucceeded, ContainerStateTerminated, "==", 0, false},
+				}
+
+				for _, testStatus := range testStatuss {
+					tmpFile, err := ioutil.TempFile("/tmp", "restartCount")
```

Comment:
Thanks all the comments.

I have not find the clear explanation of TempDir. I'm afraid that the test will fail in a system which has a different TempDir. So in the VolumeSource part, I use os.TempDir() instead of hard code ('/tmp').